inst/include/population_dynamics/fleet/fleet.hpp

```diff
@@ -142,81 +171,134 @@ namespace fims_popdy {
             // for(size_t fleet_ = 0; fleet_ <= this->nfleets; fleet_++) {
             // this -> Fmort[fleet_] = fims_math::exp(this -> log_Fmort[fleet_]);
 
-            // derived quantities
-            std::fill(catch_at_age.begin(), catch_at_age.end(),
-                    0); /**<derived quantity catch at age*/
-            std::fill(catch_index.begin(), catch_index.end(),
-                    0); /**<derived quantity catch index*/
-            std::fill(age_composition.begin(), age_composition.end(), 
-                    0); /**<model expected number at age */
-            std::fill(length_composition.begin(), length_composition.end(), 
-                    0); /**<model expected number at length */
-            std::fill(expected_catch.begin(), expected_catch.end(),
-                    0); /**<model expected total catch*/
-            std::fill(expected_index.begin(), expected_index.end(),
-                    0); /**<model expected index of abundance*/
-            std::fill(log_expected_index.begin(), log_expected_index.end(),
-                    0); /**<model expected index of abundance*/
-            std::fill(catch_numbers_at_age.begin(), catch_numbers_at_age.end(),
-                    0); /**<model expected catch at age*/
-            std::fill(proportion_catch_numbers_at_age.begin(), proportion_catch_numbers_at_age.end(),
-                    0); /**<model expected catch at age*/
-            std::fill(catch_numbers_at_length.begin(), catch_numbers_at_length.end(),
-                    0); /**<model expected catch at length*/
-            std::fill(proportion_catch_numbers_at_length.begin(), proportion_catch_numbers_at_length.end(),
-                    0); /**<model expected catch at length*/
-            std::fill(catch_weight_at_age.begin(), catch_weight_at_age.end(),
-                    0); /**<model expected weight at age*/
-
             for (size_t i = 0; i < this->log_q.size(); i++) {
                 this->q[i] = fims_math::exp(this->log_q[i]);
             }
 
             for (size_t year = 0; year < this->nyears; year++) {
                 this->Fmort[year] = fims_math::exp(this->log_Fmort[year]);
             }
+
+            // derived quantities
+            // landings
+            std::fill(landings_weight.begin(), landings_weight.end(),
+                    0); /**<model landings in weight*/
+            std::fill(landings_numbers.begin(), landings_numbers.end(),
```

Comment:
are the weight and numbers also "expected"? If so, should everything that is a derived quantity have _expected for consistency?

Reply:
So at one point I did make all the derived quantities expected but then that also got confusing. for landings and index you actually have three values one in weight , one in numbers, and one expected. The evaluate landings and evaluate index functions check which units the observed data are in and then use either weight or numbers to fill the expected vector. I figured this allows users to always be able to get expected values from one place but also see the results in numbers or weight.